src/CMakeLists.txt

```diff
@@ -15,15 +26,20 @@ target_compile_definitions(journald
         -DG_LOG_DOMAIN="eventd-journald")
 target_link_libraries(journald
     PRIVATE
-        ${EventdEvent_LIBRARIES}
+        ${Eventd_LIBRARIES}
         ${EventdPlugin_LIBRARIES}
+        ${EVENTDHELPERS_LIBRARY}
         ${SystemdJournal_LIBRARIES})
 set_target_properties(journald
     PROPERTIES
         PREFIX    ""
         NO_SONAME 1)
 
+pkg_config_variable(libeventd-plugin pluginsdir EVENTD_PLUGINS_DIR_INTERNAL)
+set(EVENTD_PLUGINS_DIR ${EVENTD_PLUGINS_DIR_INTERNAL}
```

Comment:
Ok, so at that point, we are left with the `prefix` vs. `exec-prefix` thing that, from my experience, CMake does not support (even with `GNUInstallDirs`).

With Autotools, it is easy to set `--prefix=/usr --exec-prefix=/usr/<arch>` and get all the stuff at the right place. With CMake, there is only one prefix, and with `GNUInstallDirs`, you still have to cheat (btw, setting `LIB_SUFFIX` to `/<arch>` is cheating too ;-) ).

So the safest way to check that is probably to check something longer against one of `GNUInstallDirs`’ `CMAKE_INSTALL_FULL_*` variable.
We can check:
  - `libdir` in `libeventd.pc` (arch-dependent) against `CMAKE_INSTALL_FULL_LIBDIR`
  - `datadir` in `eventd.pc` (arch-independent) against `CMAKE_INSTALL_FULL_DATADIR`
  - maybe `exec_prefix` in `libeventd.pc` against `CMAKE_INSTALL_PREFIX`

but not `prefix` (arch-independent) against `CMAKE_INSTALL_PREFIX` (arch-dependent by design I’d say)